### DISCUSSION TITLE
How do I add the processing power of my graphics card?

### DISCUSSION THREAD

**Discussion by medicinakhv** [bookfere/Ebook-Translator-Calibre-Plugin#132] [Q&A] [answered]
How do I add the processing power of my graphics card?

**Comment by bookfere** [comment]
I think there's no need to utilize GPU power since the plugin is primarily I/O-bound.

**Reply by medicinakhv** [reply]
Using the GPU in subtitle translations in Subtitle Edit https://github.com/SubtitleEdit/subtitleedit/releases allows you to increase speed by 1,000%

**Reply by bookfere** [reply, marked as answer]
I reviewed the code of the project you shared. I couldn't locate the GPU-related code in the "Auto-translate" feature, which processes pure text. Instead, I found it in the "Audio to text" feature, which automatically generates text from audio in a video.

I maintain the same opinion I expressed earlier: the plugin is I/O-bound, with a greater focus on data retrieval than data processing. In this context, GPU power cannot be effectively utilized.

**Reply by anartigone** [reply]
Subtitle Edit added [Whisper support](https://github.com/SubtitleEdit/subtitleedit/blob/cb493f0d6f741c0156ec332bdb03e7f4b5a78dd8/Changelog.txt#L171) to [utilize GPU](https://github.com/Const-me/Whisper) which is based on [whisper](https://github.com/ggerganov/whisper.cpp) and exclusively used for speech recognition. 
The so called "Auto-translate" feature uses public translation APIs which works exactly as same as ETCP here. Therefore, SE's GPU usage has nothing to do with language translation but audio to text.
Some of my workflow involves editing SRT subtitles and Subtitle Edit is my main tool for that. If you work with subtitles as well, you will be able to use your GPU power for that. There is no reason to use GPU for ebook translation, unless you run a [LLM](https://github.com/Hannibal046/Awesome-LLM#open-llm) locally as translation engine.
